Sources/eppo/RuleEvaluator.swift

```diff
@@ -48,57 +49,73 @@ public class RuleEvaluator {
         _ condition: TargetingCondition
     ) throws -> Bool
     {
-        if let value = subjectAttributes[condition.attribute] {
-            do {
-                switch condition.targetingOperator {
-                    case .GreaterThanEqualTo:
-                        return try Compare.compareNumber(
-                            value.doubleValue(),
-                            condition.value.doubleValue(),
-                            { (a: Double, b: Double) in return a >= b }
-                        );
-                    case .GreaterThan:
-                        return try Compare.compareNumber(
-                            value.doubleValue(),
-                            condition.value.doubleValue(),
-                            { (a: Double, b: Double) in return a > b }
-                        )
-                    case .LessThanEqualTo:
-                        return try Compare.compareNumber(
-                            value.doubleValue(),
-                            condition.value.doubleValue(),
-                            { (a: Double, b: Double) in return a <= b }
-                        )
-                    case .LessThan:
-                        return try Compare.compareNumber(
-                            value.doubleValue(),
-                            condition.value.doubleValue(),
-                            { (a: Double, b: Double) in return a < b }
-                        )
-                    case .Matches:
-                        return try Compare.compareRegex(
-                            value.stringValue(),
-                            condition.value.stringValue()
-                        )
-                    case .OneOf:
-                        return try Compare.isOneOf(
-                            value.stringValue(),
-                            condition.value.arrayValue()
-                        )
-                    case .NotOneOf:
-                        return try !Compare.isOneOf(
-                            value.stringValue(),
-                            condition.value.arrayValue()
-                        )
-                    default:
-                        throw Errors.UnexpectedValue
+        guard let value = subjectAttributes[condition.attribute] else {
+           return false
+        }
+        
+        do {
```

Comment:
Do we need this outer `do` now that we use a guard clause? Or can we just un-nest everything one level?

Reply:
oh sorry didn't realize that! I thought it was tied to the removed `if` statement. Appologies for my lack of swift fluency. Carry on!